Proposition change template ES6

Peltoche opened the pull request:
- Add babel config in client brunch config
- Move from old module system to the new one
- Move class attributes in contructor (specs need)
- Set some getters / setters to be Backbone and ES6 compliant

Peltoche committed:
Proposition change template ES6

- Add babel config in client brunch config
- Move from old module system to the new one
- Move class attributes in contructor (specs need)
- Set some getters / setters to be Backbone and ES6 compliant

nono commented:
@jsilvestre what do you think?

jsilvestre commented on client/app/views/app_view.es6:
```diff
@@ -1,12 +1,24 @@
   }
 
-  afterRender() {
+  afterRender () {
```
Why is there a space here?

jsilvestre commented on client/app/application.es6:
```diff
@@ -13,5 +12,5 @@ module.exports = {
-  },
-};
+  }
+}
```
I am not sure this deserved a class. It's just a function.

jsilvestre commented:
I would recommend against using classes when they are not necessary, since it's a bit of a controversial feature after all. No strong feeling though, your call @nono.

nono commented:
It's OK for me. It's a bit overkill to use a class for a simple case like this, but it shows how to do it for more complex cases (like several other features of the template).

Peltoche committed:
Add some modifications

- Forgot some module.exports
- Move some attributes in constructors
- Correct some spacing

Peltoche committed:
Corrections

- Remove some module.exports
- Move some attributes in constructors
- lint compliant

nono commented on client/app/lib/base_view.es6:
```diff
@@ -1,12 +1,12 @@
 
   getRenderData() {
-    return { model: this.model.toJSON() };
+    return { model: JSON.stringify(this.model) };
```
@Peltoche you are sure about this change?
Backbone models saves its attributes in an hidden field and use getter/setter to access them. If you do a JSON.stringify, I think you'll have all the things nested in an `attributes` object. Correct me if I am wrong.

Peltoche replied:
Oups, that's right... After consieration it seem not be the best idea to automatically send the object data so I will follow the JS template and return null by default.

nono commented on client/app/lib/view_collection.es6:
```diff
@@ -14,10 +14,18 @@ const BaseView = require('lib/base_view');
 
-  views: {}
+    this.views = {};
+    this.views = {};
```
@Peltoche it's two times the same line!

nono commented:
@Peltoche did you look at my comments?

Peltoche committed:
Fix some bugs following @nono comments
